framework/src/Volo.Abp.Ddd.Domain/Microsoft/Extensions/DependencyInjection/ServiceCollectionRepositoryExtensions.cs

```diff
@@ -80,15 +81,29 @@ public static class ServiceCollectionRepositoryExtensions
         IServiceCollection services,
         Type serviceType,
         Type implementationType,
-        bool replaceExisting)
+        bool replaceExisting,
+        bool isReadOnlyRepository = false)
     {
+        var descriptor = ServiceDescriptor.Transient(serviceType, implementationType);
```

Comment:
Instead of unnecessarily create a descriptor in case of readonly, move this code to else of the `isReadOnlyRepository` control. Here, just declare without assigning a value.